065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,298 @@
+
+This is useful with bulk replicas change, which can happen in some edge cases.
+A potentially big backlog of changes may be created when the Topic Operator is not running, or there is an issue with Cruise Control or Kafka.
+Another example is a GitOps pipeline delivering many replicas changes at the same time to reduce the storage occupation of non-critical topics.
+
+```sh
+$ curl -vv -X POST -H "Content-Type: application/json" -d '{replication_factor:{topic_by_replication_factor:{2:topic1|topic3|topic4,3:topic2|topic5}}}' \
+  "http://localhost:9090/kafkacruisecontrol/topic_configuration?skip_rack_awareness_check=true&dryrun=false&json=true" | jq
+> POST /kafkacruisecontrol/topic_configuration?skip_rack_awareness_check=true&dryrun=false&json=true HTTP/1.1
+> Host: localhost:9090
+> User-Agent: curl/8.0.1
+> Accept: */*
+> 
+} [91 bytes data]
+< HTTP/1.1 200 OK
+< Date: Thu, 07 Dec 2023 14:58:55 GMT
+< Set-Cookie: JSESSIONID=node0lfkx183ijq4pjkkjahsegfmp30.node0; Path=/
+< Expires: Thu, 01 Jan 1970 00:00:00 GMT
+< User-Task-ID: 5344ca89-351f-4565-8d0f-9aade00e053d
+< Content-Type: application/json;charset=utf-8
+< Cruise-Control-Version: 2.5.77-SNAPSHOT
+< Cruise-Control-Commit_Id: 535ea07e8847990cc02dde1f1be99b387dbeaf5b
+< Content-Length: 17051
+< Server: Jetty(9.4.52.v20230823)
+< 
+{ [17321 bytes data]
+{
+  "summary": {
+    "numIntraBrokerReplicaMovements": 0,
+    "numReplicaMovements": 125,
+    "onDemandBalancednessScoreAfter": 100,
+    "intraBrokerDataToMoveMB": 0,
+    "monitoredPartitionsPercentage": 100,
+    "provisionRecommendation": "",
+    "excludedBrokersForReplicaMove": [],
+    "excludedBrokersForLeadership": [],
+    "provisionStatus": "RIGHT_SIZED",
+    "onDemandBalancednessScoreBefore": 100,
+    "recentWindows": 5,
+    "dataToMoveMB": 0,
+    "excludedTopics": [],
+    "numLeaderMovements": 0
+  }
+  ...
+```
+
+The `skip_rack_awareness_check` parameter configures whether to skip the rack awareness sanity check or not (default to false).
+If rack awareness is not enabled in Strimzi, the `skip_rack_awareness_check` will be set to true.
+
+The `goals` parameter is a comma separated list of goals used to generate the automatic cluster rebalance for the replication factor change.
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.goals` property.
+
+The `replication_throttle` parameter is an upper bound on the bandwidth used to move replicas (bytes per second).
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.replication.throttle` property.
+
+### User tasks
+
+The [`user_tasks`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-user-request-result) endpoint will be used to periodically check the state of requested tasks.
+The only required parameter is the `User-Task-ID`, which is returned by the `topic_configuration` response.
```

Comment:
You should mention that this endpoint supports batching (`user_task_ids` is a comma-separated list).

Reply:
I did at some point. Let me add it back.